### DISCUSSION TITLE
Json codecs for Compound types

### DISCUSSION THREAD

**Discussion by Quafadas** [disneystreaming/smithy4s#1320] [General]
I have a case, where I want to decode a sub schema - `List[CfnDigraphNode]`

`CfnDigraphNode` is a union type. 

I know the machinery exists, because I can decode a class which contains this structure. If I try to access it directly so:

`val nodeJson = smithy4s.json.Json.writePrettyString[List[CfnDigraphNode]](nodes)`

yields

```No given instance of type smithy4s.schema.Schema[List[ils.smithy.ilsop.CfnDigraphNode]] was found for an implicit parameter of method writePrettyString in object Json```

**Comment by Quafadas** [comment]
```


import com.github.plokhotnyuk.jsoniter_scala.core.JsonCodec
import smithy4s.schema.Schema

val noMagic: Schema[List[CfnDigraphNode]] = Schema.list(CfnDigraphNode.schema)

val nodeJson = smithy4s.json.Json.writePrettyString[List[CfnDigraphNode]](nodes)(noMagic)
```
Appears to do the trick...

**Reply by kubukoz** [reply]
This is the way, unless there's a smithy `list` shape for `CfnDigraphNode`.

For example, if you have

```smithy
list CfnDigraphNodes {
  member: CfnDigraphNode
}
```

then you can do

```scala
val nodes: List[CfnDigraphNode] = ???

Json.writePrettyString(CfnDigraphNodes(nodes))
```
